test/battle/move_effect/retaliate.c

```diff
@@ -0,0 +1,137 @@
+        PLAYER(SPECIES_WYNAUT) { HP(1); }
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(opponent, MOVE_TACKLE); SEND_OUT(player, 1); }
+        TURN { MOVE(player, MOVE_RETALIATE); }
+        TURN { MOVE(player, MOVE_RETALIATE); }
+    } SCENE {
+        HP_BAR(opponent, captureDamage: &damage[0]);
+        HP_BAR(opponent, captureDamage: &damage[1]);
+    } THEN {
+        EXPECT_MUL_EQ(damage[1], Q_4_12(2), damage[0]);
+    }
+}
+
+SINGLE_BATTLE_TEST("Retaliate doubles in base power the turn after an ally faints (opponent)")
+{
+    s16 damage[2];
+    GIVEN {
+        PLAYER(SPECIES_WOBBUFFET);
+        OPPONENT(SPECIES_WYNAUT) { HP(1); }
+        OPPONENT(SPECIES_WOBBUFFET);
+    } WHEN {
+        TURN { MOVE(player, MOVE_TACKLE); SEND_OUT(opponent, 1); }
+        TURN { MOVE(opponent, MOVE_RETALIATE); }
+        TURN { MOVE(opponent, MOVE_RETALIATE); }
+    } SCENE {
+        HP_BAR(player, captureDamage: &damage[0]);
+        HP_BAR(player, captureDamage: &damage[1]);
+    } THEN {
+        EXPECT_MUL_EQ(damage[1], Q_4_12(2), damage[0]);
+    }
+}
+
+DOUBLE_BATTLE_TEST("Retaliate works with passive damage")
+{
+    s16 damage[2];
+    u32 move;
+    u32 move2 = MOVE_CELEBRATE;
+    struct BattlePokemon *moveTarget = playerLeft;
+    PARAMETRIZE { move = MOVE_TOXIC; moveTarget = playerLeft; }
+    PARAMETRIZE { move = MOVE_POISON_POWDER; moveTarget = playerLeft; }
+    PARAMETRIZE { move = MOVE_WILL_O_WISP; moveTarget = playerLeft; }
+    #if B_USE_FROSTBITE == TRUE
+    PARAMETRIZE { move = MOVE_ICE_BEAM; moveTarget = playerLeft; }
+    #endif
+    PARAMETRIZE { move = MOVE_SANDSTORM; moveTarget = playerLeft; }
+    PARAMETRIZE { move = MOVE_HAIL; moveTarget = playerLeft; }
+    PARAMETRIZE { move = MOVE_LEECH_SEED; moveTarget = playerLeft; }
+    PARAMETRIZE { move = MOVE_MAGMA_STORM; moveTarget = playerLeft; }
+    PARAMETRIZE { move = MOVE_FLAME_BURST; moveTarget = playerRight; }
+    PARAMETRIZE { move = MOVE_FIRE_PLEDGE; moveTarget = playerRight; move2 = MOVE_GRASS_PLEDGE; }
+    GIVEN {
+        ASSUME(gMovesInfo[MOVE_TOXIC].effect == EFFECT_TOXIC);
+        ASSUME(gMovesInfo[MOVE_POISON_POWDER].effect == EFFECT_POISON);
+        ASSUME(gMovesInfo[MOVE_WILL_O_WISP].effect == EFFECT_WILL_O_WISP);
+        #if B_USE_FROSTBITE == TRUE
+        ASSUME(gMovesInfo[MOVE_ICE_BEAM].additionalEffects[0].moveEffect == MOVE_EFFECT_FREEZE_OR_FROSTBITE);
+        #endif
+        ASSUME(gMovesInfo[MOVE_SANDSTORM].effect == EFFECT_SANDSTORM);
+        ASSUME(gMovesInfo[MOVE_HAIL].effect == EFFECT_HAIL);
+        ASSUME(gMovesInfo[MOVE_LEECH_SEED].effect == EFFECT_LEECH_SEED);
+        ASSUME(gMovesInfo[MOVE_MAGMA_STORM].additionalEffects[0].moveEffect == MOVE_EFFECT_WRAP);
+        ASSUME(gMovesInfo[MOVE_FLAME_BURST].additionalEffects[0].moveEffect == MOVE_EFFECT_FLAME_BURST);
+        PLAYER(SPECIES_WYNAUT) { Ability(ABILITY_SHADOW_TAG); HP(18); Defense(9999); SpDefense(9999); }
```

Comment:
Is there a way to make the test legal? Maybe an opponent with a level 1 mon?

Reply:
Probably, I kinda just took the definition from what Edu did with the all moves test, since I wrote this during that time. I can check in a bit.

Reply:
I see. It is probably better to be consistent and make the tests legal where possible.

Reply:
Fixed